src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/TypeLoaderExports.cs

```diff
@@ -7,11 +7,72 @@
 using System.Threading;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
+using System.Numerics;
 
 namespace System.Runtime
 {
+    // Initialize the cache eagerly to avoid null checks.
+    [EagerStaticClassConstruction]
     public static class TypeLoaderExports
     {
+        //
+        // Generic lookup cache
+        //
+
+#if DEBUG
+        // use smaller numbers to hit resizing/preempting logic in debug
+        private const int InitialCacheSize = 8; // MUST BE A POWER OF TWO
+        private const int MaximumCacheSize = 512;
+#else
+        private const int InitialCacheSize = 128; // MUST BE A POWER OF TWO
+        private const int MaximumCacheSize = 128 * 1024;
+#endif // DEBUG
+
+        private static GenericCache<Key, Value> s_cache =
+            new GenericCache<Key, Value>(InitialCacheSize, MaximumCacheSize);
+
+        private struct Key : IEquatable<Key>
+        {
+            public IntPtr _context;
+            public IntPtr _signature;
+
+            public Key(nint context, nint signature)
+            {
+                _context = context;
+                _signature = signature;
+            }
+
+            public bool Equals(Key other)
+            {
+                return _context == other._context && _signature == other._signature;
+            }
+
+            public override int GetHashCode()
+            {
+                // pointers will likely match and cancel out in the upper bits
+                // we will rotate context by 16 bit to keep more varying bits in the hash
+                IntPtr context = (IntPtr)BitOperations.RotateLeft((nuint)_context, 16);
+                return (context ^ _signature).GetHashCode();
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj is Key && Equals((Key)obj);
+            }
+        }
+
+        private struct Value
+        {
+            public IntPtr _result;
+            public IntPtr _auxResult;
+
+            public Value(IntPtr result, IntPtr auxResult)
+            {
+                _result = result;
+                _auxResult = auxResult;
+            }
+        }
+
         public static unsafe void ActivatorCreateInstanceAny(ref object ptrToData, IntPtr pEETypePtr)
```

Comment:
This method looks like a dead code. Delete it?

Reply:
It does seem unused.